src/discovery/azure/Akka.Discovery.Azure/AzureServiceDiscovery.cs

```diff
@@ -28,13 +28,19 @@ public class AzureServiceDiscovery : ServiceDiscovery
 
         private readonly IActorRef _guardianActor;
 
+        // Backward compatibility constructor
         public AzureServiceDiscovery(ExtendedActorSystem system)
+            : this(system, system.Settings.Config.GetConfig("akka.discovery.azure"))
+        {
+        }
+        
+        public AzureServiceDiscovery(ExtendedActorSystem system, Configuration.Config config)
         {
             _system = system;
             _log = Logging.GetLogger(system, typeof(AzureServiceDiscovery));
             
-            _system.Settings.InjectTopLevelFallback(DefaultConfig);
-            _settings = AzureDiscoverySettings.Create(system);
+            var fullConfig = config.WithFallback(DefaultConfig.GetConfig("akka.discovery.azure"));
+            _settings = AzureDiscoverySettings.Create(system, fullConfig);
             
             var setup = _system.Settings.Setup.Get<AzureDiscoverySetup>();
```

Comment:
This line is currently problematic, a single registered `AzureDiscoverySetup` instance can pollute every instance of this ServiceDiscovery instance. Would need to look at how Akka.Persistence handles this.

Reply:
Fixed this using a hack by embedding the discovery id inside the Akka.Hosting generated HOCON config. We can map/corellate which config goes to which setup class this way.